lib/kafkalib/topic.go

```diff
@@ -65,20 +66,24 @@ func (t *TopicConfig) String() string {
 		t.Database, t.Schema, t.TableName, t.Topic, t.IdempotentKey, t.CDCFormat, t.DropDeletedColumns)
 }
 
-func (t *TopicConfig) Valid() bool {
+func (t *TopicConfig) Validate() error {
 	if t == nil {
-		return false
+		return fmt.Errorf("topic config is nil")
 	}
 
 	// IdempotentKey is optional.
 	empty := array.Empty([]string{t.Database, t.Schema, t.Topic, t.CDCFormat})
 	if empty {
-		return false
+		return fmt.Errorf("database, schema, topic or cdc format is empty")
 	}
 
 	if t.CDCKeyFormat == "" {
 		t.CDCKeyFormat = defaultKeyFormat
```

Comment:
This is a little gross, the `Validate` method shouldn't be mutating the topic config. Maybe add a method for this as a FF?

Reply:
Agreed, chatted offline. Will do a FF